tests/samples/matmul_fill_pack_config.mlir

```diff
@@ -0,0 +1,31 @@
+// This test is used to check the IR generated from AMDAIE passes to be the same as
+// what generated from transform dialect (matmul_fill_spec_pack_peel.mlir).
+
+// RUN: iree-opt %s --pass-pipeline="builtin.module(hal.executable(hal.executable.variant(iree-amdaie-lower-workgroup-count, cse, builtin.module(func.func(iree-amdaie-tile-and-fuse{tiling-level=0}, iree-amdaie-tile-and-fuse{use-scf-for tiling-level=1}, iree-amdaie-cleanup, canonicalize, cse, iree-amdaie-pack-and-transpose{pack-level=1}, iree-amdaie-bufferize-to-allocation{memory-space=1}, iree-amdaie-tile-and-fuse{tiling-level=2}, iree-amdaie-cleanup, canonicalize, cse, iree-amdaie-pack-and-transpose{pack-level=2}, iree-amdaie-bufferize-to-allocation{memory-space=2}, iree-hoist-statically-bound-allocations, iree-amdaie-peel-for-loop, iree-amdaie-cleanup, canonicalize, cse)))), iree-eliminate-empty-tensors, iree-codegen-iree-comprehensive-bufferize, canonicalize, cse, canonicalize)"
```

Comment:
This PR is trying to put passes in the same order as in [matmul_fill_spec_pack_peel.mlir](https://github.com/nod-ai/iree-amd-aie/blob/main/tests/samples/matmul_fill_spec_pack_peel.mlir). What you tried is a simplified version of packing pipeline. I have a WIP patch that can work with Erwei's patches for the simplified version.

Reply:
I see, can you please share the WIP patch so that @Abhishek-Varma and I can try it out and figure out next work for the OPT GEMMs. We will not try the large sizes but just the different types (transposes, BMM's and so on) so that in case more passes are needed to allow that we can start working on those.

Reply:
Yes, I'll clean up and push the patch tomorrow